### PR TITLE
Feature/network interface

### LINKED ISSUE #43: nifcloud_network_interface: Add resource
### Overview

Add a new resource `nifcloud_network_interface`.

### PR DESCRIPTION
**Pull request by fuku2014**
## Description

- Add nifcloud_network_interface resource.
- Closes #43  

## How Has This Been Tested?

- [ ]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/network_interface
terraform init
terraform plan
terraform apply
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/resources/networkinterface/expander_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+)
+
+func TestExpandCreateNetworkInterfaceInput(t *testing.T) {
+	rd := schema.TestResourceDataRaw(t, newSchema(), map[string]interface{}{
+		"ip_address":        "test_ip_address",
+		"network_id":        "test_network_id",
+		"description":       "test_description",
+		"availability_zone": "test_availability_zone",
+	})
+	rd.SetId("test_network_interface_id")
```
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/networkinterface/expander_test.go]: Fixed!

File: nifcloud/resources/networkinterface/expander.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+func expandModifyNetworkInterfaceAttributeInputForIPAddress(d *schema.ResourceData) *computing.ModifyNetworkInterfaceAttributeInput {
+	return &computing.ModifyNetworkInterfaceAttributeInput{
+		NetworkInterfaceId: nifcloud.String(d.Id()),
+		IpAddress:          nifcloud.String(d.Get("ip_address").(string)),
+	}
+}
+
+func expandDeleteNetworkInterfaceInput(d *schema.ResourceData) *computing.DeleteNetworkInterfaceInput {
+	input := &computing.DeleteNetworkInterfaceInput{
```
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/networkinterface/expander.go]: Fixed!

File: nifcloud/resources/networkinterface/expander.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		Description:    nifcloud.String(d.Get("description").(string)),
+		Placement: &computing.RequestPlacementOfCreateNetworkInterface{
+			AvailabilityZone: nifcloud.String(d.Get("availability_zone").(string)),
+		},
+	}
+	return input
+}
+
+func expandDescribeNetworkInterfacesInput(d *schema.ResourceData) *computing.DescribeNetworkInterfacesInput {
+	input := &computing.DescribeNetworkInterfacesInput{
```
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/networkinterface/expander.go]: Fixed!

File: nifcloud/resources/networkinterface/expander.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package networkinterface
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/nifcloud/nifcloud-sdk-go/nifcloud"
+	"github.com/nifcloud/nifcloud-sdk-go/service/computing"
+)
+
+func expandCreateNetworkInterfaceInput(d *schema.ResourceData) *computing.CreateNetworkInterfaceInput {
+	input := &computing.CreateNetworkInterfaceInput{
```
[the remark this answers is not in the document]
Comment (fuku2014) [reply nifcloud/resources/networkinterface/expander.go]: Fixed!

**Review by tunakyonn — APPROVED** [review]
LGTM!

**Pull request merged**